allergies/allergies.py

```diff
@@ -13,30 +13,18 @@
 class Allergies(object):
 
     def __init__(self, score):
-        self.score = score
-        self.allergen_list = self.lst
+        self.allergen_list = []
+        self.score = score % 256
 
-#    @property
+    @property
     def lst(self):
-        allergen_list = []
-        score = self.score % 256
         factors = 7
         while factors >= 0:
-            if score >= 2**factors:
-                score -= 2**factors
-                allergen_list.insert(0, ALLERGENS.get(2**factors))
+            if self.score >= 2**factors:
+                self.score -= 2**factors
+                self.allergen_list.insert(0, ALLERGENS.get(2**factors))
             factors -= 1
-        return allergen_list
+        return self.allergen_list
 
     def is_allergic_to(self, item):
-        self.item = item
-
-        allergen_list = []
-        score = self.score % 256
-        factors = 7
-        while factors >= 0:
-            if score >= 2**factors:
-                score -= 2**factors
-                allergen_list.append(ALLERGENS.get(2**factors))
-            factors -= 1
-        return item in allergen_list
+        return item in self.lst
```

Comment:
Ok, so here we can improve the performance a bit. Right now you are re-calculating the list each time (by calling `self.lst`). At the same time, you are keeping the result of the calculations in `allergen_list`. How about trying to check here first if the `self.allergen_list` has already been calculated and looking up if `item` is there? And if it hasn't yet been generated - call self.lst and then check if the `item` is there. (Hint: I'd start with `self.allergen_list = None` in `__init__` to have a value that means "not generated yet" as opposed to `[]` which would mean "generated but no matches found")

Reply:
@virtuallight ok, i get it that i generate `allergen_list `while running `lst`, so i came up with 
```
    def is_allergic_to(self, item):
        if self.allergen_list == []:
            self.allergen_list = self.lst
        return item in self.allergen_list
```
but when i turn `self.allergen_list = None` in `__init__` i get an error
`AttributeError: 'NoneType' object has no attribute 'append' `
`TypeError: argument of type 'NoneType' is not iterable `
`AssertionError: None != []`
like it don't know that it is a list... have no idea hwo to implement it right.

Reply:
@zarniak Those errors that you are getting seem to stem from now having done `self.allergen_list = []` at the very beginning of `def lst(self):`. So, once you reach `self.allergen_list.append` the `self.allergen_list` is still `None`.